<launch>
    <arg name="mavros" default="false"/>
    <arg name="simple_offboard" default="false"/>

    <arg name="vive_base_frame_id" default="map"/>
    <arg name="static_yaw" default="-1.5707963"/>
    <arg name="static_pitch" default="0.0"/>
    <arg name="static_roll" default="3.1415926"/>

    <!-- mavros -->
    <include file="$(find roslibsurvive)/launch/mavros.launch" if="$(arg mavros)">
        <arg name="fcu_conn" default="usb"/>
        <arg name="fcu_ip" default="127.0.0.1"/>
        <arg name="gcs_bridge" default="tcp"/>
        <arg name="viz" default="false"/>
        <arg name="respawn" default="true"/>
    </include>
    
    <!-- vive tracker pose sender -->
    <node name="frame_publisher" pkg="roslibsurvive" type="roslibsurvive_frame_publisher">
        <param name="base_frame_id" value="$(arg vive_base_frame_id)"/>       
        <param name="yaw_static" value="$(arg static_yaw)"/>
        <param name="pitch_static" value="$(arg static_pitch)"/>
        <param name="roll_static" value="$(arg static_roll)"/>   
        <remap from="~T20_pose_t1" to="/mavros/vision_pose/pose"/>      
    </node>

    <!-- simple offboard for copter control -->
    <node name="simple_offboard" pkg="clever" type="simple_offboard" output="screen" clear_params="true" if="$(arg simple_offboard)">
        <param name="reference_frames/body" value="map"/>
        <param name="reference_frames/base_link" value="map"/>
    </node>

    <!-- Auxiliary frames -->
    <node name="frames" pkg="clever" type="frames" output="screen">
        <param name="body/frame_id" value="body"/>
    </node>

</launch>

